components/Page.tsx

```diff
@@ -34,7 +44,7 @@ export default function Page(props) {
         <link rel="shortcut icon" href="/favicon.ico" />
       </Head>
       <img className={styles.pixel} src="https://intdev-global.s3.us-west-2.amazonaws.com/template-app-icon.png" alt="" />
-      {props.children}
+      <ViewerContext.Provider value={props.viewer}>{props.children}</ViewerContext.Provider>
```

Comment:
Thought: I get why you would want to make this optimization. But now all components that are children that have viewer will expect to receive it through context, instead of the natural way which is passing through props. This consideration is lost here but I don't feel so strongly to debate it here.

Verdict: Seems fine to introduce this model as a way we develop code and be more opinionated.

Reply:
my original thought was to pass the viewer as a prop when the modal is opened. however, since props are only passed in when the modal is opened, any changes will not cause it to update. the best solution i found was to just put a context as an ancestor to the `GlobalModalManager` (in `Page` in this case), which means that changes to the viewer will be reflected in any modals that need it in real time.

an alternate approach i thought of just now is we could add a function called something like `modal.updateProps(props)`. this would allow modifying props of a modal in-place without closing / reopening it. if the modal was not open then it would be a no-op. this would solve the problem and we could avoid this context. does that sound good?

Reply:
I think thats more predictable, if you fall in love with `updateProps` go with that, otherwise we'll live with the viewer as context variable, and have to explain that if you use the modal components you must have a viewer context setup.

Reply:
i looked at the code and found that there are currently no use cases where props for an open modal actually need to be updated. i can of course see this being needed in the future, but i'm going to opt to do this in a followup since it's a non-breaking addition, so we can get this pr merged.